.clj-kondo/config.edn

```diff
@@ -745,6 +747,15 @@
      toucan.util.test {:message "Don't use application database inside MLv2 code"}
      toucan2.core     {:message "Don't use application database inside MLv2 code"}}}}
 
+  non-api-namespaces
+  {:linters
+   {:discouraged-var
+    {metabase.api.common/*current-user*                 {:message "Only use API-related dynamic variables in the API namespaces"}
```

Comment:
Would it make more sense to do `:discouraged-namespace` for `metabase.api.common` instead? I think you could actually maybe even create a namespace pattern like `metabase\\.api\\..*` and use that in `:discouraged-namespaces`.

Using `metabase.api.user/some-helper-function` inside `metabase.models.permissions-group` or something like that is hardly any better than using the dynamic variables

Reply:
That makes sense. We could use the [:ns-groups](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#discouraged-namespace) feature.